includes/data-stores/abstract-wc-order-data-store-cpt.php

```diff
@@ -108,6 +108,14 @@ public function read( &$order ) {
 				$order->set_discount_total( (double) get_post_meta( $order->get_id(), '_cart_discount', true ) - (double) get_post_meta( $order->get_id(), '_cart_discount_tax', true ) );
 			}
 		}
+
+		/**
+		 * In older versions, paid date may not have been set.
+		 * @todo When/if meta is flattened, handle this in the migration script.
+		 */
+		if ( ! $order->get_version( 'edit' ) || version_compare( $order->get_version( 'edit' ), '2.7', '<' ) ) {
+			$order->maybe_set_date_paid( $order->get_date_created( 'edit' ) );
```

Comment:
I think this should check the order version and if there is no `date_paid` value set.

Otherwise, it will override any existing `date_paid` value based only on the order version. It's possible even if the order's version < 2.7 that the order has an existing and valid `date_paid` value set that is different to the `date_created`. In those cases, that value should be honoured.